md_rahaman/python/django/ninja_gold/apps/ninja_gold_app/views.py

```diff
@@ -0,0 +1,69 @@
+from django.shortcuts import render,redirect
+import os
+from random import randint
+import datetime
+
+# Create your views here.
+
+
+def index(request):
+	if 'gold' not in request.session:
+		request.session['gold'] = 0
+		request.session['money'] = 0
+		request.session['activities'] = []
+  	return render(request, 'ninja_gold_app/index.html')
+
+def process_money(request):
```

Comment:
This method contains a lot of code duplication. Instead of repeating the assignment of request.session values in each if/elif condition. Set the session key after you've done all your logic, which will remove the code duplication that you have.